src/SAWScript/Crucible/LLVM/Override.hs

```diff
@@ -1315,18 +1328,18 @@ invalidateMutableAllocs opts sc cc cs = do
 
   -- set of (concrete base pointer, size) for each postcondition memory write
   postPtrs <- Set.fromList <$> mapM
-    (\(LLVMPointsTo _loc ptr val) -> do
+    (\(LLVMPointsTo _loc cond ptr val) -> do
       (_, Crucible.LLVMPointer blk _) <- resolveSetupValue opts cc sc cs Crucible.PtrRepr ptr
       sz <- (return . Crucible.storageTypeSize)
         =<< Crucible.toStorableType
         =<< typeOfSetupValue cc (MS.csAllocations cs) (MS.csTypeNames cs) val
-      return (W4.asNat blk, sz))
+      return (W4.asNat blk, sz, isJust cond))
     (cs ^. MS.csPostState ^. MS.csPointsTos)
 
   -- filter out each allocation overwritten by a postcondition write
   let danglingPtrs = filter
         (\((Crucible.LLVMPointer blk _), sz, _) ->
-          Set.notMember (W4.asNat blk, sz) postPtrs)
+          Set.notMember (W4.asNat blk, sz, False) postPtrs)
```

Comment:
Again, to make sure I've understood: for the purposes of memory invalidation, this only treats unconditional points-to as ones that don't require an invalidation pass, right?

Is it worth instead combining postcondition conditional points-to into a single mux memory operation that does the write under the condition and does the invalidate otherwise?  I guess the semantics are basically the same, but I wonder if there are performance implications.

Reply:
I wonder about this, too. I also wonder whether @weaversa's suggestions on making conditional points-to total (i.e., an if-then-else rather than just an if-then) would simplify or complicate this check.

Reply:
the conditional points-to and the invalidation are combined: https://github.com/GaloisInc/saw-script/pull/657/commits/f3c8e7bc612d3f328d738f29aa68fc55476d517f